Author: Jaldhar H. Vyas <jaldhar@debian.org>
Description: Small cosmetic changes for dovecot-sql.conf

--- dovecot-1.2.7.orig/doc/dovecot-sql-example.conf
+++ dovecot-1.2.7/doc/dovecot-sql-example.conf
@@ -47,7 +47,7 @@
 #                           the default my.cnf location
 #     option_group        - Read options from the given group (default: client)
 # 
-#   You can connect to UNIX sockets by using host: host=/var/run/mysql.sock
+#   You can connect to UNIX sockets by using host: host=/var/run/mysqld/mysqld.sock
 #   Note that currently you can't use spaces in parameters.
 #
 #   MySQL supports multiple host parameters for load balancing / HA.
