Description: Start dovecot even if protocols is none
Origin: vendor
--- dovecot-1.2.7.orig/src/master/master-settings.c
+++ dovecot-1.2.7/src/master/master-settings.c
@@ -851,6 +851,7 @@
 	}
 
 	if (set->protocol != MAIL_PROTOCOL_ANY &&
+	    strcmp(set->protocols, "none") &&
 	    access(t_strcut(set->mail_executable, ' '), X_OK) < 0) {
 		i_error("mail_executable: Can't use %s: %m",
 			t_strcut(set->mail_executable, ' '));
@@ -949,6 +950,7 @@
 	}
 
 	if (set->protocol != MAIL_PROTOCOL_ANY &&
+	    strcmp(set->protocols, "none") &&
 	    access(t_strcut(set->login_executable, ' '), X_OK) < 0) {
 		i_error("login_executable: Can't use %s: %m",
 			t_strcut(set->login_executable, ' '));
