Author: Stephan Bosch <stephan@rename-it.nl>
Description: Deprecated imapflags extension: fixed implicit assignment of flags.
Origin: http://hg.rename-it.nl/dovecot-1.2-sieve/rev/a890258aa5a9
Last-Update: 2010-02-16

--- a/sieve/src/lib-sieve/plugins/imap4flags/ext-imap4flags-common.c
+++ b/sieve/src/lib-sieve/plugins/imap4flags/ext-imap4flags-common.c
@@ -149,16 +149,18 @@
 
 void ext_imap4flags_attach_flags_tag
 (struct sieve_validator *valdtr, const struct sieve_extension *ext,
-	const char *command)
+	const char *command, bool implicit)
 {
 	/* Register :flags tag with the command and we don't care whether it is
 	 * registered or even whether it will be registered at all. The validator
 	 * handles either situation gracefully
 	 */
 
-	/* Tag specified by user */
-	sieve_validator_register_external_tag
-		(valdtr, command, ext, &tag_flags, SIEVE_OPT_SIDE_EFFECT);
+	if ( !implicit ) {
+		/* Tag specified by user */
+		sieve_validator_register_external_tag
+			(valdtr, command, ext, &tag_flags, SIEVE_OPT_SIDE_EFFECT);
+	}
 
     /* Implicit tag if none is specified */
 	sieve_validator_register_persistent_tag
--- a/sieve/src/lib-sieve/plugins/imap4flags/ext-imap4flags-common.h
+++ b/sieve/src/lib-sieve/plugins/imap4flags/ext-imap4flags-common.h
@@ -68,7 +68,7 @@
 
 void ext_imap4flags_attach_flags_tag
 	(struct sieve_validator *valdtr, const struct sieve_extension *ext,
-		const char *command);
+		const char *command, bool implicit);
 
 /*
  * Flag management
--- a/sieve/src/lib-sieve/plugins/imap4flags/ext-imap4flags.c
+++ b/sieve/src/lib-sieve/plugins/imap4flags/ext-imap4flags.c
@@ -68,8 +68,9 @@
 	sieve_validator_register_command(valdtr, ext, &cmd_removeflag);
 	sieve_validator_register_command(valdtr, ext, &tst_hasflag);
 
-	ext_imap4flags_attach_flags_tag(valdtr, ext, "keep");
-	ext_imap4flags_attach_flags_tag(valdtr, ext, "fileinto");
+	/* Attach :flags tag to keep and fileinto commands */
+	ext_imap4flags_attach_flags_tag(valdtr, ext, "keep", FALSE);
+	ext_imap4flags_attach_flags_tag(valdtr, ext, "fileinto", FALSE);
 
 	return TRUE;
 }
--- a/sieve/src/lib-sieve/plugins/imap4flags/ext-imapflags.c
+++ b/sieve/src/lib-sieve/plugins/imap4flags/ext-imapflags.c
@@ -131,6 +131,10 @@
 	sieve_validator_register_command(valdtr, master_ext, &cmd_mark);
 	sieve_validator_register_command(valdtr, master_ext, &cmd_unmark);
 
+    /* Attach implicit flags tag to keep and fileinto commands */
+    ext_imap4flags_attach_flags_tag(valdtr, master_ext, "keep", TRUE);
+    ext_imap4flags_attach_flags_tag(valdtr, master_ext, "fileinto", TRUE);
+
 	return TRUE;
 }
 
