From: Noah Meyerhans <noahm@debian.org>
Date: Thu, 21 May 2020 21:48:59 2020 -0700
Subject: Don't try to build doc/rfc subdir components
---

Index: dovecot/pigeonhole/configure.ac
===================================================================
--- dovecot.orig/pigeonhole/configure.ac
+++ dovecot/pigeonhole/configure.ac
@@ -176,7 +176,6 @@ doc/Makefile
 doc/man/Makefile
 doc/example-config/Makefile
 doc/example-config/conf.d/Makefile
-doc/rfc/Makefile
 doc/extensions/Makefile
 doc/locations/Makefile
 doc/plugins/Makefile
Index: dovecot/pigeonhole/doc/Makefile.am
===================================================================
--- dovecot.orig/pigeonhole/doc/Makefile.am
+++ dovecot/pigeonhole/doc/Makefile.am
@@ -1,7 +1,6 @@
 SUBDIRS = \
 	man \
 	example-config \
-	rfc \
 	extensions \
 	locations \
 	plugins
Index: dovecot/pigeonhole/doc/Makefile.in
===================================================================
--- dovecot.orig/pigeonhole/doc/Makefile.in
+++ dovecot/pigeonhole/doc/Makefile.in
@@ -399,7 +399,6 @@ top_srcdir = @top_srcdir@
 SUBDIRS = \
 	man \
 	example-config \
-	rfc \
 	extensions \
 	locations \
 	plugins
